go/launcher/proxy/driverhub/driver_hub.go

```diff
@@ -124,12 +126,42 @@ func (h *WebDriverHub) GetActiveSessions() []string {
 	return result
 }
 
-// Shutdown is a no-op on WebDriverHub.
+// Shutdown is shutdowns any running sessions.
 func (h *WebDriverHub) Shutdown(ctx context.Context) error {
-	// Shutdown sessions
+	for _, id := range h.GetActiveSessions() {
+		session := h.GetSession(id)
+		session.quit(ctx, false)
+	}
+	for _, session := range h.reusableSessions {
+		session.quit(ctx, false)
+	}
 	return nil
 }
 
+// GetReusableSession grabs a reusable session if one is available that matches caps.
+func (h *WebDriverHub) GetReusableSession(caps map[string]interface{}) (*WebDriverSession, bool) {
+	if !capabilities.CanReuseSession(caps) {
+		return nil, false
+	}
+
+	h.mu.Lock()
+	defer h.mu.Unlock()
+	for i, session := range h.reusableSessions {
+		if capabilities.Equals(caps, session.Desired) {
+			h.reusableSessions = append(h.reusableSessions[:i], h.reusableSessions[i+1:]...)
+			return session, true
+		}
+	}
+	return nil, false
+}
+
+func (h *WebDriverHub) AddReusableSession(session *WebDriverSession) {
+	if !capabilities.CanReuseSession(session.Desired) {
+		return
```

Comment:
Seems like it should be an error or something if you attempt to add a non-reusable session here. I suppose you currently would never use the return value, though.